src/eth_provider/provider.rs

```diff
@@ -615,9 +622,13 @@ where
         // Get the number of retries for the transaction
         let retries = self.database.pending_transaction_retries(&transaction_signed.hash).await?;
 
+        let transaction_signed_ec_recovered =
+            TransactionSignedEcRecovered::from_signed_transaction(transaction_signed.clone(), signer);
+
+        let encoded_length = transaction_signed_ec_recovered.clone().length_without_header();
+
         // Upsert the transaction as pending in the database
-        let transaction =
-            from_recovered(TransactionSignedEcRecovered::from_signed_transaction(transaction_signed.clone(), signer));
+        let transaction = from_recovered(transaction_signed_ec_recovered.clone());
         self.database.upsert_pending_transaction(transaction, retries).await?;
```

Comment:
or are we keeping this in order to split the mempool PR in multiple small PR's?

Reply:
We don't need retry anymore but I think we should keep it for now and rm in follow up otherwise it will break tests and stuffs and the PR will be huge